stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -138,15 +147,33 @@ internal class PaymentOptionsActivity : BasePaymentSheetActivity<PaymentOptionRe
                 }
             }
         }
+    }
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
+    @VisibleForTesting
+    private val viewStateObserver: (ViewState.PaymentOptions?) -> Unit
+        get() = { viewState ->
+            lifecycleScope.launch {
+                val addButton = viewBinding.addButton
+                addButton.updateState(
+                    when (viewState) {
+                        null -> null
+                        is ViewState.PaymentOptions.Ready -> PrimaryButton.State.Ready(addButtonLabel)
+                        is ViewState.PaymentOptions.Confirming -> PrimaryButton.State.Confirming
+                        is ViewState.PaymentOptions.Completed -> PrimaryButton.State.Completed
+                    }
+                )
+
+                if (viewState is ViewState.PaymentOptions.Completed) {
+                    onActionCompleted(viewState.result)
+                }
```

Comment:
Won't this close the sheet before animation is completed?

Reply:
Not if animate is blocking - see below.

Reply:
By making the animation blocking in the primary button, we don't have to send the viewState result into the primary button.  This is important because the result is a different type based on if you are in the PaymentOptions or the PaymentSheet Complete.  

I could pass in callback, but this reads way more iteratively which is one of the awesome things about kotlin coroutines!

Reply:
I can switch it to a callback for now.